agent/acl_endpoint_test.go

```diff
@@ -592,5 +645,251 @@ func TestACL_HTTP(t *testing.T) {
 			require.Len(t, token.Policies, 1)
 			require.Equal(t, structs.ACLPolicyGlobalManagementID, token.Policies[0].ID)
 		})
+		t.Run("Create with Accessor", func(t *testing.T) {
+			tokenInput := &structs.ACLToken{
+				AccessorID:  "56e8e6a3-708b-4a2f-8ab3-b973cce39108",
+				Description: "test",
+				Policies: []structs.ACLTokenPolicyLink{
+					structs.ACLTokenPolicyLink{
+						ID:   idMap["policy-test"],
+						Name: policyMap[idMap["policy-test"]].Name,
+					},
+					structs.ACLTokenPolicyLink{
+						ID:   idMap["policy-read-all-nodes"],
+						Name: policyMap[idMap["policy-read-all-nodes"]].Name,
+					},
+				},
+			}
+
+			req, _ := http.NewRequest("PUT", "/v1/acl/token?token=root", jsonBody(tokenInput))
+			resp := httptest.NewRecorder()
+			obj, err := a.srv.ACLTokenCreate(resp, req)
+			require.NoError(t, err)
+
+			token, ok := obj.(*structs.ACLToken)
+			require.True(t, ok)
+
+			// 36 = length of the string form of uuids
+			require.Equal(t, tokenInput.AccessorID, token.AccessorID)
```

Comment:
This block of `require.XXX` calls appears several times. Maybe consider making a local verification function to use in the call sites instead to make the test easier to skim?

Reply:
Most are slightly different. Some verify the accessor is the one specified and just that the secret is 36 bytes. Some verify the secret and length of accessor. Some verify both. 

I think line item equality is better than creating a verification function to account for all the different scenarios. IMO test code should be as simple as possible. You don't want to need to test the tests if you can help it.